### PR TITLE
add responsive menu

### LINKED ISSUE #63: add responsive menu
### Motivation
Currently the menu is not responsive, it's not easy to use in small screen size.

### Proposed Solution
Add a responsive menu

### PR DESCRIPTION
**Pull request by noworneverev**
### Short description
Add responsive menu

### Proposed changes
Automatically minimize the the menu in small screen
- 
- 

### Side effects

- 
- 

### Resolved issues

Fixes: #63

### CODE REVIEW

File: ycms/static/src/js/utils/menu.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+window.addEventListener("load", () => {
+    const menuToggle = document.getElementById("menu-toggle");
+    const mobileMenu = document.getElementById("mobile-menu");
+
+    menuToggle?.addEventListener("click", () => {
+        mobileMenu?.classList.toggle("hidden");
```
Comment (charludo) [inline ycms/static/src/js/utils/menu.ts]: `hidden` cannot be animated, sadly. How about `-translate-x-full`?

File: ycms/static/src/css/style.scss [hunk 2]
```diff
@@ -278,3 +301,31 @@ input[type="submit"] {
...
+#mobile-menu li a.selected {
+    @apply pl-6 text-gray-800 bg-gray-200 border-blue-500;
+}
+
+#mobile-menu li a svg {
+    @apply w-[22px] mr-2;
+}
+
+#menu-toggle {
+    transition: all 0.3s ease;
```
Comment (charludo) [inline ycms/static/src/css/style.scss:314]: I think you want to apply the transition duration to the mobile menu / sidebar itself, right?

File: ycms/cms/templates/_hamburger.html [hunk 3]
```diff
@@ -0,0 +1,89 @@
+{% load i18n %}
+{% load static %}
+<!-- Hamburger menu for smaller screens -->
```
Comment (charludo) [inline ycms/cms/templates/_hamburger.html]: I appreciate the effort. However, there is really no need for a second menu. You can just apply the classes/styles/animations... to the existing sidebar, so we don't have duplicate structures.

File: ycms/cms/templates/_base.html [hunk 4]
```diff
@@ -53,89 +53,93 @@
...
-                 height="auto"
-                 alt="{% translate " Logo" %}"
-                 class="my-3 mx-auto">
-        </a>
-        <ul>
-            <span class="block mt-5 px-5 text-gray-500 tracking-wide">{% translate "General" %}</span>
-            {% if perms.cms.add_ward %}
+    <div class="lg:flex">
+        <div id="sidebar"
+             class="fixed h-full w-64 bg-white border-r shadow-md hidden lg:block">
```
Comment (charludo) [inline ycms/cms/templates/_base.html]:
```suggestion
             class="fixed h-full w-64 bg-white border-r shadow-md transform -translate-x-full lg:translate-x-0">
```
This way we can use the one sidebar on desktop and mobile, plus `translate-x` is animateable! :D

**Review by charludo — CHANGES_REQUESTED** [review]
Thanks, this is a very good idea, and I'm impressed how good the site (esp. the ward view 😄) looks on mobile with this!

I have two thoughts:
- as mentioned in the comments below, there's no need for a second, mobile menu, since it just copies the structures from the desktop menu. Please just apply the classes/animations to the `sidebar`.
- Could you maybe also swap out the icon of the menu on click? Just would be a nice touch I think :D E.g. from icon name `menu` to `x` or something like that.

**Review by charludo — APPROVED** [review]
Very cool, thank you!

Only nitpick I have is it would be cool if the menu was still full height and the button would move to the right with it, but this absolutely works. SO if you have an idea for that feel free to implement it, otherwise go ahead and merge 😄

**Comment by noworneverev** [comment]
> Very cool, thank you!
> 
> Only nitpick I have is it would be cool if the menu was still full height and the button would move to the right with it, but this absolutely works. SO if you have an idea for that feel free to implement it, otherwise go ahead and merge 😄

I think it's enough for now, we can improve it later.

**Pull request merged**